src/openrct2/interface/Viewport.cpp

```diff
@@ -1260,12 +1261,12 @@ void viewport_set_visibility(uint8_t mode)
  * Originally checked 0x0141F569 at start
  *  rct2: 0x00688697
  */
-static void store_interaction_info(paint_struct* ps)
+static std::optional<InteractionInfo> GetInteractionInfoFromPS(paint_struct* ps, uint16_t flags)
 {
     if (ps->sprite_type == VIEWPORT_INTERACTION_ITEM_NONE
         || ps->sprite_type == 11 // 11 as a type seems to not exist, maybe part of the typo mentioned later on.
         || ps->sprite_type > VIEWPORT_INTERACTION_ITEM_BANNER)
-        return;
+        return {};
```

Comment:
Thoughts on returning this instead?
```suggestion
        return std::nullopt;
```